app/src/main/java/invoice_automation/module/QuickBooksModule.java

```diff
@@ -83,8 +84,34 @@ public List<Customer> getAllCustomers() {
      * @return A copy of the Customer updated or created
      */
     public Customer updateCustomerFromSchool(@NonNull School school) {
-        // TODO: IA-7
-        return null;
+        boolean addNew = true;
+        // Get customer associated with the school, if it exists
+        Customer schoolCustomer = new Customer();
+        List<Customer> customers = getAllCustomers();
+        for (Customer c: customers) {
+            if (c.getDisplayName().equals(school.getSchoolName())) {
+                schoolCustomer = c;
+                addNew = false;
+                break;
+            }
+        }
+        // Convert the school object to a customer object
+        Customer newCustomer = QuickBooksUtils.getCustomerFromSchool(school);
+        if (!addNew) {
+            newCustomer.setId(schoolCustomer.getId());
+            newCustomer.setSyncToken(schoolCustomer.getSyncToken());
+        }
+        // Add newCustomer or update the existing customer
+        try {
+            if (addNew) {
+                dataService.add(newCustomer);
+            } else {
+                dataService.update(newCustomer);
```

Comment:
can you do sparse update:
<img width="856" alt="Screenshot 2023-08-21 at 16 26 01" src="https://github.com/skberkeley/bmun_invoicing/assets/66887530/5c532bca-deba-4ba4-a266-b325f7b6abfc">